src/fsinfo.ts

```diff
@@ -113,33 +136,17 @@ export function fsinfo(path: string, attrs: string[], options?: cockpit.JsonObje
         }
         return null; // ELOOP
     }
+}
 
-    const channel = cockpit.channel({
-        superuser: "try",
-        payload: "fsinfo",
-        path,
-        attrs,
-        "close-on-error": false,
-        watch: true,
-        ...options
-    });
-
-    let state: cockpit.JsonValue = null;
-    channel.addEventListener("message", (_event: CustomEvent, payload: string) => {
-        state = json_merge(state, JSON.parse(payload));
-
-        if (is_json_dict(state) && !state.partial) {
-            self.state = {
-                info: is_json_dict(state.info) ? state.info : null,
-                error: is_json_dict(state.error) ? state.error : null,
-                loading: false
-            };
-
-            for (const callback of callbacks) {
-                callback(self.state);
+export function fsinfo(path: string, attrs: (keyof FileInfo)[], options?: cockpit.JsonObject): Promise<FileInfo> {
+    return new Promise((resolve, reject) => {
```

Comment:
This looks like a backwards compat function, but the code in app.tsx was already ported to use `FsInfoClient` directly. This is also rather shallow -- do we need this for anything outside of c-files?

Reply:
This is not backcompat but is intended as a non-compatible API that does something different: a one-shot (watch: false) query of the current state, which it simply returns, or an exception.

I started writing a Qunit test for this on a branch in the cockpit repo...

Reply:
I see, ok! It's fine to keep it here then, I was just curious.